core/src/main/scala/org/apache/spark/ui/exec/ExecutorsTab.scala

```diff
@@ -138,10 +138,12 @@ class ExecutorsListener(storageStatusListener: StorageStatusListener, conf: Spar
           // could have failed half-way through. The correct fix would be to keep track of the
           // metrics added by each attempt, but this is much more complicated.
           return
-        case _: ExceptionFailure =>
-          taskSummary.tasksFailed += 1
         case _ =>
-          taskSummary.tasksComplete += 1
+      }
+      if (info.successful) {
+        taskSummary.tasksComplete += 1
+      } else {
+        taskSummary.tasksFailed += 1
```

Comment:
I went through the list of possible task failure reasons, and I found info.successful to be updated for all except Resubmitted case. If you think there is a possibility of any other reason that has been overlooked by me, do let me know.